rundeck/rundeck
# RUN-2221: Update webpack 4 to 5

**Is this a bugfix, or an enhancement? Please describe.**
-



List of enhancement/maintanance:

- Update webpack 4 to 5;
- Update packages related to vue-cli and jest;
- Remove/fix/update imports that had issues - some were wrong, some aren't needed anymore;
- Remove commands that were necessary only for webpack 4;
- Small fixes (especially syntax) that would prevent builds or tests from running;


**Describe the solution you've implemented**
-


Followed the guide https://webpack.js.org/migrate/5/  and the following order for updating:

updated webpack > other npm packages > errors that would prevent the project from running/building locally > tests /imports/syntax errors 

Checks performed:
- ran final image with docker;
- ran locally all commands used by the project;


How to test:
- 

- run this branch locally, firstly npm install then run project's `core` settings (can also do `enterprise` as well, but then please remember to run installUiTrellis first);
- check for any missing files or errors in the UI;

Or test with a docker image;

### **Describe alternatives you've considered**


In the past there have been efforts to move the FE to vite, but as it requires a bigger effort to move out from vue-cli+webpack to vite, updating webpack to the newest version was a compromise for improving build speed/security and also ensuring compatibility with new npm packages that require webpack 5.

**Additional context**

Added comments explaining what was done in this PR. 

Also, it might be worth to consider to use other modes of devtools as it greatly improves build time (One run, which had other modes enabled had a performance gain of 40% https://app.circleci.com/pipelines/github/rundeck/rundeck/14735)

**_Final performance improvement on circle ci: 5%_**
